spec/support/shared_examples/interaction_tracking.rb

```diff
@@ -0,0 +1,34 @@
+RSpec::Matchers.define_negated_matcher :not_change, :change
+
+shared_examples_for "an incoming interaction" do
+  it "updates the associated client" do
+    expect { subject.save }
+      .to change(subject.client, :attention_needed_since)
+      .and change(subject.client, :last_incoming_interaction_at)
+      .and change(subject.client, :last_interaction_at)
+      .and change(subject.client, :updated_at)
+  end
+end
```

Comment:
This doesn't seem to cover the case when a client already needs attention. It looks like if a client kept messaging us, each time it would reset attention_needed_since and last_incoming_interaction_at.